packages/headless/src/utils/highlight.ts

```diff
@@ -150,11 +150,20 @@ function suggestionWithDelimiters(
  */
 
 export function escape(str: string) {
-  return str
-    .replace(/&/g, '&amp')
-    .replace(/</g, '&lt')
-    .replace(/>/g, '&gt')
-    .replace(/"/g, '&quot')
-    .replace(/`/g, '&#96')
-    .replace(/'/g, '&#x27');
+  const mapOfCharToEscape: Record<string, string> = {
```

Comment:
If possible, I would prefer to avoid having to maintain our own map of characters. If this map of characters, as used by Lodash, is later determined to be a vulnerability, we won't be notified since we aren't using Lodash's own function.

It looks like [`encodeURIComponent`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent) encodes all those characters except the apostrophe, which shouldn't be a problem for HTML, but I don't see any guarantee that the output of `encodeURIComponent` is innerHTML-safe.

I think we should use a library for this.

Reply:
> It looks like [encodeURIComponent](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent) encodes all those characters except the apostrophe, which shouldn't be a problem for HTML, but I don't see any guarantee that the output of encodeURIComponent is innerHTML-safe.

encodeURIComponent is to encode characters to be used in URL. It's not really related to what we are really interested here.

> I think we should use a library for this.

I disagree. 

I get your first point about security, but I think it's a bit over-engineered. 

The chances that a new HTML characters will magically appear one day and introduce a problem is... low 😄 .

To illustrate differently, Lodash implementation has not changed in 6 years (which is pretty much the age of the library).